satpy/readers/__init__.py

```diff
@@ -16,31 +16,33 @@
 # You should have received a copy of the GNU General Public License along with
 # satpy.  If not, see <http://www.gnu.org/licenses/>.
 """Shared objects of the various reader classes."""
+from __future__ import annotations
 
 import logging
 import os
+import pickle
 import warnings
 from datetime import datetime, timedelta
 from functools import total_ordering
-import pickle
 
 import yaml
 
 try:
     from yaml import UnsafeLoader
 except ImportError:
-    from yaml import Loader as UnsafeLoader
+    from yaml import Loader as UnsafeLoader  # type: ignore
```

Comment:
What does the comment `# type: ignore` do?  Is some linter going to complain about this line otherwise?

Reply:
Mypy was complaining. I don't know why though

Reply:
Yes, mypy complained and is still complaining in other modules with:

```
Incompatible import of "UnsafeLoader" (imported name has type "Type[Loader]", local name has type "Type[UnsafeLoader]")
```

I'm not sure there is a way around that.